src/protocolgame.cpp

```diff
@@ -988,6 +988,23 @@ void ProtocolGame::parseRotateItem(NetworkMessage& msg)
 	addGameTaskTimed(DISPATCHER_TASK_EXPIRATION, &Game::playerRotateItem, player->getID(), pos, stackpos, spriteId);
 }
 
+void ProtocolGame::parseRuleViolationReport(NetworkMessage &msg)
+{
+	uint8_t reportType = msg.getByte();
+	uint8_t reportReason = msg.getByte();
+	const std::string targetName = msg.getString();
+	const std::string comment = msg.getString();
+	std::string translation;
```

Comment:
@gugahoa client actually send 3 different datas in this same byte, reports for unofficial client only write report type, reason and comment, report due player name write type, reason, comment and translation, player statement everything plus a statement id, it's the client buffer for message position, so i believe we need these ifs for sync with server socket buffer.